agent/tcs/client/client.go

```diff
@@ -315,6 +359,19 @@ func (cs *clientServer) createPublishHealthRequests() ([]*ecstcs.PublishHealthRe
 	return requests, nil
 }
 
+// createPublishInstanceHealthRequests creates the requests to publish instance health
+func (cs *clientServer) createPublishInstanceHealthRequests() *ecstcs.PublishInstanceHealthRequest {
```

Comment:
There's a possibility here of metrics being lost between calling the Get* functions and running the deferred ResetCount function.

I'd suggest having some function(s) that do an atomic "Swap" to both return the current value and reset the count at the same time. You can use https://golang.org/pkg/sync/atomic/#SwapInt64 for this.

It might be a good idea to just change ResetCount to do the swap and return the 'old' values. I think the interface would be safer this way because I don't think there is currently a use-case where we would want to reset the counts without also getting them. ie instead of having a ResetCount function we could have two functions: GetAndResetCallCount() and GetAndResetErrorCount() or a single GetAndResetCount() that returns the two counts.

Reply:
Thanks for pointing this. Updated!